### PR TITLE
Open source hyperbahn.

### PR DESCRIPTION
**Pull request by Raynos**
There is a lot here because hyperbahn is a lot.

All the tests are here and they pass. The Makefile
is here and it works.

Our closed source hyperbahn will just be a bunch
of integration with uber specific infrastructure
like alerting and deploying.

This depends on a closed source code review of
the internal hyperbahn before getting merged.

r: @jcorbin @rf @kriskowal

### CODE REVIEW

File: hyperbahn/bin/admin.js [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+
+function Admin(options) {
+    if (!(this instanceof Admin)) {
+        return new Admin(options);
+    }
+
+    var self = this;
+
+    self.hosts = options.hosts || getHostPortsForProd(
+        '/etc/uber/hyperbahn/ringpop-v2.json'
```
Comment (abhinav) [inline hyperbahn/bin/admin.js]: ?

Comment (Raynos) [inline hyperbahn/bin/admin.js]: This should use config instead... hmm.

**Comment by anson627** [comment]
why not standalone hyperbahn repo? Does it mean git clone, npm install, npm test etc. are going to be slower?

**Comment by jcorbin** [comment]
The plan is to split it out once the existing hyperbahn folder is
completed, which is what this PR does

**Comment by Raynos** [comment]
We need to consolidate `hyperbahn` and `node/hyperbahn` into one folder. 

This is a multi-step progress where we try to break as few open branches as possible; one step at a time.

**Comment by Raynos** [comment]
People can review at their leisure. This is going in. I'm going to release the lock on these thousands of lines of code by migrating it to their new home.

**Pull request merged**